### PR TITLE
virtcontainers: support vhost and physical endpoints in AddInterface

### LINKED ISSUE #758: network: AddInterface only supports VirtualEndpoints
If we look at the way `AddInterface()` has been implemented, it will always create a `virtualEndpoint`, no matter the type of device backing the interface that we're trying to add. Unfortunately, this means neither `vhost-user` nor `vfio` can be leveraged in this case.

This needs to be fixed either by adding a new field `devType` to the `grpc.Interface` structure passed as an argument of `AddInterface()`, or by adding a new argument `devType` directly to the function itself.

@bergwolf @caoruidong @WeiZhang555 @amshinde WDYT?

### PR DESCRIPTION
**Pull request by caoruidong**
Support to hotplug vhost and physical endpoints by AddInterface API

Fixes #758

Signed-off-by: Ruidong Cao <caoruidong@huawei.com>

### CODE REVIEW

File: virtcontainers/network.go [hunk 1]
```diff
@@ -212,6 +215,11 @@ func (endpoint *VirtualEndpoint) Type() EndpointType {
 	return endpoint.EndpointType
 }
 
+// PciAddr returns the PCI address of the endpoint.
+func (endpoint *VirtualEndpoint) PciAddr() string {
```
Comment (devimc) [inline virtcontainers/network.go]: `s/PciAddr/PCIAddr` ?

**Comment by jcvenegas** [comment]
CI failing on `vm template tests`  https://github.com/kata-containers/tests/issues/760 not related with this PR.

File: virtcontainers/sandbox.go [hunk 2]
```diff
@@ -17,8 +17,8 @@ import (
 	"syscall"
 
 	"github.com/containernetworking/plugins/pkg/ns"
-	specs "github.com/opencontainers/runtime-spec/specs-go"
```
Comment (jcvenegas) [inline virtcontainers/sandbox.go]: I think this change is not needed, is this an error from a formatting tool?

File: virtcontainers/sandbox.go [hunk 3]
```diff
@@ -17,8 +17,8 @@ import (
...
-	opentracing "github.com/opentracing/opentracing-go"
```
Comment (jcvenegas) [inline virtcontainers/sandbox.go]: Same comment, do this changed needed ?

File: virtcontainers/sandbox.go (hunk 2 above)
Comment (caoruidong) [reply virtcontainers/sandbox.go]: Yes. I will discard this.

File: virtcontainers/network.go (hunk 1 above)
Comment (caoruidong) [reply virtcontainers/network.go]: That will be same with field name and makes an error.

File: virtcontainers/sandbox.go [hunk 4]
```diff
@@ -1105,7 +1105,32 @@ func (s *Sandbox) AddInterface(inf *grpc.Interface) (*grpc.Interface, error) {
 		return nil, err
 	}
 
-	endpoint, err := createVirtualNetworkEndpoint(len(s.networkNS.Endpoints), inf.Name, s.config.NetworkConfig.InterworkingModel)
+	var endpoint Endpoint
```
Comment (bergwolf) [inline virtcontainers/sandbox.go]: This chunk of code is copied from part of `createEndpointsFromScan`. Please add a new function for it in network.go and just call the new function here.

Comment (caoruidong) [reply virtcontainers/sandbox.go]: Done. There is a closure.

**Comment by jcvenegas** [comment]
@devimc  @bergwolf  take a look again :)

**Comment by sboeuf** [comment]
@bergwolf @amshinde PTAL

**Pull request merged**